tests/publiccloud/upload_image.pm

```diff
@@ -31,14 +32,16 @@ sub run {
         return;
     }
 
-    # Download the given image. Check for 404 errors and make them better visible
-    my $cmd = "wget --no-check-certificate $img_url -O $img_name";
-    my $rc = script_run("$cmd 2>download.txt", timeout => 60 * 10);
+    # Download the given image via wget. Note that by default wget retries 20 times before giving up
+    my $cmd = "wget --no-check-certificate --retry-connrefused --retry-on-host-error $img_url -O $img_name";
+    # A generious timeout is required because downloading up to 30 GB (Azure images) can take more than an hour.
+    my $rc = script_run("$cmd 2>&1 | tee download.txt", timeout => 120 * 60);
     if ($rc != 0) {
-        upload_logs("download.txt");
-        script_run("cat download.txt");
-        die "404 - Image not found" if (script_run("grep '404' download.txt") == 0);
-        die "command '$cmd' failed with rc=$rc";
+        # Check for 404 errors and make them better visible
+        my $output = script_output("cat download.txt");
+        record_info("wget failed with status code $rc", "$cmd\n\n$output");
```

Comment:
```suggestion
        record_info("wget finished with status code $rc", "$cmd\n\n$output");
```

Reply:
Since this is in the `if ($rc != 0)` branch I would much rather keep it at failed.